### PR TITLE
Avoid and remove some SSL error codes for Mbed TLS 3.0

### PR DESCRIPTION
**Pull request by hanno-arm**
__Context:__ This PR continues #4446 in pursuit of cleaning up the SSL error code space for Mbed TLS 3.0.

__Dependency:__ Based on #4446.

__This PR:__
* Replaces the single use of `MBEDTLS_ERR_SSL_CERTIFICATE_REQUIRED` by `MBEDTLS_ERR_SSL_INTERNAL_ERROR` (https://github.com/ARMmbed/mbedtls/commit/9cfe6e977d14bdc56f80e6f06a757ca7ccd42da2) and removes `MBEDTLS_ERR_SSL_CERTIFICATE_REQUIRED` (https://github.com/ARMmbed/mbedtls/commit/3268d84313e16297b4f21ab251d5c5436e72d04c)
* Removes internal-only uses of `MBEDTLS_ERR_SSL_INVALID_VERIFY_HASH` (https://github.com/ARMmbed/mbedtls/commit/6c780469609da63915ec9930efb2a8ee0985799c) and removes the latter from the public error space (https://github.com/ARMmbed/mbedtls/commit/56ee9e5f141bb5cf3d0468fa09b6de7f5b19daf6).
* Replaces the single use of `MBEDTLS_ERR_SSL_CERTIFICATE_TOO_LARGE` by `MBEDTLS_ERR_SSL_BUFFER_TOO_SMALL` and removes `MBEDTLS_ERR_SSL_CERTIFICATE_TOO_LARGE` from the public error space (https://github.com/ARMmbed/mbedtls/commit/91e1cc3bd7dcfe89dd0c9c2d6ad7041f839c5c31)

### CODE REVIEW

**Review by davidhorstmann-arm — APPROVED** [review]

File: library/ssl_tls.c [hunk 1]
```diff
@@ -6998,14 +6998,14 @@ int mbedtls_ssl_set_calc_verify_md( mbedtls_ssl_context *ssl, int md )
 {
 #if defined(MBEDTLS_SSL_PROTO_TLS1_2)
     if( ssl->minor_ver != MBEDTLS_SSL_MINOR_VERSION_3 )
-        return MBEDTLS_ERR_SSL_INVALID_VERIFY_HASH;
+        return( 1 );
```
Comment (mpg) [inline library/ssl_tls.c:7002]: I know this doesn't change anything it practice, but I think returning `-1` would be slightly clearer. Error codes tend to be negative, and we already use `-1` is several places as a generic error code.

**Review by mpg — CHANGES_REQUESTED** [review]
Looks good to me, but as for 4446, I think brief ChangeLog and migration guide entries are in order.

File: library/ssl_tls.c (hunk 1 above)
Comment (hanno-arm) [reply library/ssl_tls.c:7002]: Changing.

**Comment by hanno-arm** [comment]
@mpg Thank you for your review. I've added a ChangeLog and Migration Guide entry. Could you take another look?

**Comment by hanno-arm** [comment]
@davidhorstmann-arm Could you re-review the migration guide and change log entries, too, please?

File: docs/3.0-migration-guide.d/ssl-error-cleanup.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+  /home/hanbec01/mbedtlsE/docs/3.0-migration-guide.d:
+  total used in directory 16 available 77914544
+  drwxrwxr-x 2 hanbec01 hanbec01 4096 May 14 16:48 .
```
Comment (hanno-arm) [inline docs/3.0-migration-guide.d/ssl-error-cleanup.md:3]: I don't know what I've done here... fixing.

**Comment by hanno-arm** [comment]
@davidhorstmann-arm @mpg I apparently messed something up in the latest commit -- fixing this first.

**Review by davidhorstmann-arm — APPROVED** [review]
LGTM - thanks!

**Review by mpg — APPROVED** [review]
LGTM too!

**Pull request merged**